pandas/core/arrays/categorical.py

```diff
@@ -1567,8 +1567,12 @@ def argsort(self, *args, **kwargs):
         >>> cat.argsort()
         array([3, 0, 1, 2])
         """
-        # Keep the implementation here just for the docstring.
-        return super().argsort(*args, **kwargs)
+        ascending = nv.validate_argsort_with_ascending(ascending, args, kwargs)
```

Comment:
why does this impl change? the _values_for_argsort is properly defined so this should *just* work no?

Reply:
The `NA-first` behaviour of `Categorical` will be changed to `NA-last` in the next PR. So the original `_values_for_argsort` is placed here temporarily.